deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1PodHandler.java

```diff
@@ -372,4 +393,122 @@ public boolean cleanUp() {
 	public int getWeight() {
 	    return ResourceKind.POD.getWeight();
 	}
+	
+	public void watch(ApiClient apiClient, String appName, String serviceName, String namespace)
+			throws IOException, ApiException, HyscaleException {
+		OkHttpClient httpClient = apiClient.getHttpClient().newBuilder().readTimeout(0, TimeUnit.SECONDS).build();
+		apiClient.setHttpClient(httpClient);
+		CoreV1Api api = new CoreV1Api(apiClient);
+
+		String selector = ResourceSelectorUtil.getServiceSelector(appName, serviceName);
+		
+		Integer replicas = null;
+		String latestPodSelector = null;
+		V1StatefulSet v1StatefulSet=null;
+		try {
+			v1StatefulSet=getV1StatefulSetHandler().get(apiClient, serviceName, namespace);
+		}catch(Exception e) {
```

Comment:
Avoid using generic exception try to be specific like HyscaleException. Log the exception in logger